### ISSUE TITLE
1.5 Error deleting draft AP Transactions

### ISSUE THREAD

**Issue by nick-prater** [ledgersmb/LedgerSMB#2421]
Having entered some draft AP Transactions, I attempted to delete them. This resulted in errors:

ERROR: update or delete on table "acc_trans" violates foreign key constraint "cr_report_max_ac_id_fkey" on table "cr_report" DETAIL: Key (entry_id)=(6425) is still referenced from table "cr_report". CONTEXT: SQL statement "DELETE FROM acc_trans WHERE trans_id = in_id" PL/pgSQL function draft_delete(integer) line 9 at SQL statement: SELECT * FROM "public"."draft_delete"(?) at /home/nick/perl5/perlbrew/perls/perl-5.24.0/lib/site_perl/5.24.0/PGObject.pm line 359.

This is similar, but different to previously fixed bug #2380, where deletes need to be cascaded for a number of tables referencing acc_trans.

PR for consideration to follow shortly.

**Comment by ehuelsmann** [comment]
Does this transaction include a reconciled payment? If so, is the
reconciliation unapproved, as of yet?

**Comment by nick-prater** [comment]
As I remember the transactions involved did not include reconciled payments - they were transactions I had just entered.

Last night I managed several times to generate this bug. But in the cold light of day, having deleted the problematic transactions, I've been unable to replicate the problem. I'm therefore closing this issue and associated PR. I'll revive it if I encounter the problem again - otherwise sorry for taking up valuable time.
